sql/src/main/java/io/crate/metadata/sys/ClassifiedHistograms.java

```diff
@@ -54,6 +57,33 @@ public Histogram histogram() {
         public Classification classification() {
             return classification;
         }
+
+        long sumOfDurations() {
+            return sumOfDurations;
+        }
+    }
+
+    private static class Metrics {
+
+        private final ConcurrentHistogram histogram;
+        private final AtomicLong sumOfDurations = new AtomicLong();
+
+        /**
+         * @param highestTrackableValue represents the highest value to track
+         * @param numberOfSignificantValueDigits the precision to use
+         */
+        Metrics(final long highestTrackableValue, final int numberOfSignificantValueDigits) {
+            histogram = new ConcurrentHistogram(highestTrackableValue, numberOfSignificantValueDigits);
+        }
+
+        void recordValue(long duration) {
+            histogram.recordValue(Math.min(Math.max(0, duration), HIGHEST_TRACKABLE_VALUE));
+            sumOfDurations.addAndGet(duration);
+        }
+
+        static ClassifiedHistogram toClassifiedHistogram(Classification classification, Metrics metrics) {
+            return new ClassifiedHistogram(metrics.histogram, classification, metrics.sumOfDurations.get());
```

Comment:
Would it be possible to expose `Metrics` directly to avoid the conversion ? (reduce allocations)